tuxedo-core/src/constraint_checker.rs

```diff
@@ -0,0 +1,111 @@
+//! A constraint checker is a piece of logic that determines whether a transaction as a whole is valid
+//! and should be committed. Most tuxedo pieces will provide one or more constraint checkers.
+//! Constraint Checkers do not typically calculate the correct final state, but rather determine whether the
+//! proposed final state (as specified by the output set) meets the necessary constraints.
+
+use sp_std::{fmt::Debug, vec::Vec};
+
+use crate::{dynamic_typing::DynamicallyTypedData, types::Output, Verifier};
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_runtime::transaction_validity::TransactionPriority;
+
+/// A single constraint checker that a transaction can choose to call. Checks whether the input
+/// and output data from a transaction meets the codified constraints.
+///
+/// Additional transient information may be passed to the constraint checker by including it in the fields
+/// of the constraint checker struct itself. Information passed in this way does not come from state, nor
+/// is it stored in state.
+pub trait SimpleConstraintChecker: Debug + Encode + Decode + Clone {
+    /// The error type that this constraint checker may return
+    type Error: Debug;
+
+    /// The actual check validation logic
+    fn check(
+        &self,
+        input_data: &[DynamicallyTypedData],
+        output_data: &[DynamicallyTypedData],
+    ) -> Result<TransactionPriority, Self::Error>;
+}
+
+pub trait ConstraintChecker: Debug + Encode + Decode + Clone {
```

Comment:
We should have some docs on this trait. Maybe the ones form the old `SimpleConstraintChecker` trait should go here and those ones be updated.

Reply:
Will do !